File: Python/2-Medium/00056._合并区间/00056._合并区间.py
from typing import List

def merge_intervals(intervals: List[List[int]]) -> List[List[int]]:
    """
    合并所有重叠的区间。

    Args:
    intervals: List[List[int]] 一个区间的集合，每个区间由两个整数表示（起始位置，结束位置）。

    Returns:
    List[List[int]] 合并后的区间列表。

    示例:
    >>> merge_intervals([[1,3],[2,6],[8,10],[15,18]])
    [[1,6],[8,10],[15,18]]

    """
    # 按照区间的起始位置进行排序
    intervals.sort(key=lambda x: x[0])

    merged = []  # 初始化合并后的区间列表
    for interval in intervals:
        # 如果合并列表为空或当前区间的起始位置大于合并列表中最后一个区间的结束位置
        # 则不重叠，将当前区间添加到合并列表中
        if not merged or merged[-1][1] < interval[0]:
            merged.append(interval)
        else:
            # 否则，有重叠，合并区间，更新最后一个区间的结束位置为当前区间和最后一个区间结束位置的最大值
            merged[-1][1] = max(merged[-1][1], interval[1])
    
    return merged

# 测试代码
test_intervals = [[1,3],[2,6],[8,10],[15,18]]
merged_intervals = merge_intervals(test_intervals)
print("合并后的区间：", merged_intervals)

File: C++/2-Medium/00056._合并区间/00056._合并区间.cpp
#include <iostream>
#include <vector>
#include <algorithm> // 用于std::sort

using namespace std;

// 合并区间的函数
vector<vector<int>> mergeIntervals(vector<vector<int>>& intervals) {
    if (intervals.empty()) return {}; // 处理空数组的情况
    
    // 按区间的起始位置排序
    sort(intervals.begin(), intervals.end(), [](const vector<int>& a, const vector<int>& b) {
        return a[0] < b[0];
    });
    
    vector<vector<int>> merged;
    for (const auto& interval : intervals) {
        // 如果merged为空或当前区间与merged中最后一个区间不重叠，直接添加
        if (merged.empty() || merged.back()[1] < interval[0]) {
            merged.push_back(interval);
        } else {
            // 否则，有重叠，合并区间
            merged.back()[1] = max(merged.back()[1], interval[1]);
        }
    }
    
    return merged;
}

int main() {
    // 示例输入
    vector<vector<int>> intervals = {{1,3},{2,6},{8,10},{15,18}};
    
    // 调用函数并打印结果
    vector<vector<int>> mergedIntervals = mergeIntervals(intervals);
    cout << "Merged intervals: ";
    for (const auto& interval : mergedIntervals) {
        cout << "[" << interval[0] << "," << interval[1] << "] ";
    }
    
    return 0;
}